app/models/jwt_secret.rb

```diff
@@ -0,0 +1,25 @@
+class JwtSecret < ApplicationRecord
+  include Encryptable
+
+  encrypts :token
+
+  belongs_to :user, inverse_of: :jwt_secret
+
+  validates :token, uniqueness: true
+  validates :user, presence: true
+
+  before_save :generate_token, on: :create, prepend: true, :unless => Proc.new{|j| j.token.present?}
+
+  def self.for_user(user_or_user_id)
```

Comment:
@timogoebel scope `for_user` returns all records if there is no record matching the user
```
[1] pry(main)> JwtSecret.for_user(User.find(1))
=> [#<JwtSecret:0x00007fded3df2190
  id: 1,
  token: "rmZbjJV3RIOQ5shPb21COQ==",
  user_id: 2,
  created_at: Tue, 12 Feb 2019 14:07:25 UTC +00:00,
  updated_at: Tue, 12 Feb 2019 14:07:25 UTC +00:00>,
 #<JwtSecret:0x00007fded3df2000
  id: 4,
  token: "8LsZmPrAwtOcu/Zi3q3idQ==",
  user_id: 5,
  created_at: Tue, 12 Feb 2019 14:09:39 UTC +00:00,
  updated_at: Tue, 12 Feb 2019 14:09:39 UTC +00:00>]
```
I suggest to go with `find_by` instead
```
[2] pry(main)> JwtSecret.find_by(user: User.find(1))
=> nil
```
If we want to keep the scope we should use `where` instead of `find_by` in the scope as scope should return `ActiveRecord::Relation` object.